src/librustdoc/clean/mod.rs

```diff
@@ -2232,6 +2232,13 @@ impl Clean<Vec<Item>> for doctree::ExternCrate<'_> {
 
 impl Clean<Vec<Item>> for doctree::Import<'_> {
     fn clean(&self, cx: &DocContext<'_>) -> Vec<Item> {
+        // We need this comparison because some imports (for std types for example)
+        // are "inserted" as well but directly by the compiler and they should not be
+        // taken into account.
+        if self.span.is_dummy() {
```

Comment:
How will this interact with proc-macros? I think they sometimes won't have spans.

cc @danielhenrymantilla

Reply:
I don't know enough if there are different kind of "dummy" spans, and regarding proc-macros, I don't know if (part) of the issue is still out there, but at some point it was indeed possible to receive (and thus emit) tokens that had lost their span information, and had a `0..0` "dummy" span.

@Aaron1011 fixed many of those span-passing errors between the compiler and proc-macros, so they should be able to provide a more concrete answer 🙂

Reply:
Fixing these issues is tracked by https://github.com/rust-lang/rust/issues/43081.

Once that issue is resolved, I don't think proc-macros should be emitting dummy spans (unless we add a way to synthesize them). The one exception might be built-in macros - some of the code in `compiler/rustc_builtin_macros` emits `DUMMY_SP`, but I don't think this is observable by proc-macros.

That being said, I think we can use a more specifi check than `is_dummy`. From reading the linked discussion, I think this is intended to handle the prelude imports. If so, you can check for `ExpnKind::AstPass(AstPass::StdImports)` on the span's outer `SyntaxContext`.

Reply:
> From reading the linked discussion, I think this is intended to handle the prelude imports.

Indeed it is! Thanks for the help :) @GuillaumeGomez can you change this to the more specific check Aaron suggested?

Reply:
Done! Thanks a lot for the tip!